From bbaab9157d6d8b4806aa603767ba4da304b62d4b Mon Sep 17 00:00:00 2001
From: Sivasubramaniyan Padmanaban <sivasubramaniyan.padmanaban@analog.com>
Date: Thu, 18 Jan 2024 16:09:30 +0530
Subject: [PATCH] drivers/media/i2c/adsd3500.c: Added MP mode RAW12 resolution.

drivers/media/spi/adsd3500-spi.c: Changed the RAW 16 to RAW8 pixel
format.

Signed-off-by: Sivasubramaniyan Padmanaban <sivasubramaniyan.padmanaban@analog.com>
---
 drivers/media/i2c/adsd3500.c     | 21 +++++++
 drivers/media/spi/adsd3500-spi.c | 94 ++++++++++++++++++++++++--------
 2 files changed, 93 insertions(+), 22 deletions(-)

diff --git a/drivers/media/i2c/adsd3500.c b/drivers/media/i2c/adsd3500.c
index 36fea3193fd4..524bb0c9c51c 100644
--- a/drivers/media/i2c/adsd3500.c
+++ b/drivers/media/i2c/adsd3500.c
@@ -379,6 +379,20 @@ static const struct adsd3500_mode_info adsd3500_mode_info_data[] = {
 		.code = MEDIA_BUS_FMT_SBGGR12_1X12,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
+	{ //RAW12 12BPP * 3 phase + 12BPP * 3 AB
+		.width = 2048,
+		.height = 5376,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR12_1X12,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
+	{ //RAW12 12BPP * 2 phase + 12BPP * 2 AB
+		.width = 2048,
+		.height = 3584,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR12_1X12,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
 	{ //RAW12 12BPP * 9 subframes ADSD3100 512x512x9
 		.width = 1024,
 		.height = 2304,
@@ -456,6 +470,13 @@ static const struct adsd3500_mode_info adsd3500_mode_info_data[] = {
 		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
+	{ //RAW8 MP 16BPP * 3 phase + 16BPP * 3 AB
+		.width = 2048,
+		.height = 6144,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
 	{ //RAW8 QMP 16BPP * 3 phase + 16BPP * 1 AB
 		.width = 1024,
 		.height = 2048,
diff --git a/drivers/media/spi/adsd3500-spi.c b/drivers/media/spi/adsd3500-spi.c
index 882790b772d3..0385302f6afd 100644
--- a/drivers/media/spi/adsd3500-spi.c
+++ b/drivers/media/spi/adsd3500-spi.c
@@ -389,6 +389,20 @@ static const struct adsd3500_mode_info adsd3500_mode_info_data[] = {
 		.code = MEDIA_BUS_FMT_SBGGR12_1X12,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
+	{ //RAW12 12BPP * 3 phase + 12BPP * 3 AB
+		.width = 2048,
+		.height = 5376,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR12_1X12,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
+	{ //RAW12 12BPP * 2 phase + 12BPP * 2 AB
+		.width = 2048,
+		.height = 3584,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR12_1X12,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
 	{ //RAW12 12BPP * 9 subframes ADSD3100 512x512x9
 		.width = 1024,
 		.height = 2304,
@@ -452,83 +466,119 @@ static const struct adsd3500_mode_info adsd3500_mode_info_data[] = {
 		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
-	{ //RAW16 MP 3 phase + 1 AB
+	{ //RAW8 MP 16BPP * 3 phase + 16BPP * 1 AB
 		.width = 2048,
 		.height = 4096,
 		.pixel_rate = 488000000,
-		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
-	{ //RAW16 MP 3 phase
+	{ //RAW8 MP 16BPP * 3 phase
 		.width = 2048,
 		.height = 3072,
 		.pixel_rate = 488000000,
-		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
+	{ //RAW8 MP 16BPP * 3 phase + 16BPP * 3 AB
+		.width = 2048,
+		.height = 6144,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
-	{ //RAW16 QMP 3 phase + 1 AB
+	{ //RAW8 QMP 16BPP * 3 phase + 16BPP * 1 AB
 		.width = 1024,
 		.height = 2048,
 		.pixel_rate = 488000000,
-		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
-	{ //RAW16 QMP 3 phase + 3 AB
+	{ //RAW8 QMP 16BPP * 3 phase + 16BPP * 3 AB
 		.width = 3072,
 		.height = 3072,
 		.pixel_rate = 488000000,
-		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
-	{ //RAW16 QMP 3 phase
+	{ //RAW8 QMP 16BPP * 3 phase
 		.width = 1024,
 		.height = 1536,
 		.pixel_rate = 488000000,
-		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
-	{ //RAW16 VGA 3 phase + 1 AB
+	{ //RAW8 VGA 16BPP * 3 phase + 16BPP * 1 AB
 		.width = 1024,
 		.height = 2560,
 		.pixel_rate = 488000000,
-		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
-	{ //RAW16 VGA 3 phase + 3 AB
+	{ //RAW8 VGA 16BPP * 3 phase + 16BPP * 3 AB
 		.width = 3072,
 		.height = 3840,
 		.pixel_rate = 488000000,
-		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
-	{ //RAW16 VGA 3 phase
+	{ //RAW8 VGA 16BPP * 3 phase
 		.width = 1024,
 		.height = 1920,
 		.pixel_rate = 488000000,
-		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
-	{ //RAW16 QVGA 3 phase + 1 AB
+	{ //RAW8 QVGA 16BPP * 3 phase + 16BPP * 1 AB
 		.width = 512,
 		.height = 1920,
 		.pixel_rate = 488000000,
-		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
-	{ //RAW16 QVGA 3 phase + 3 AB
+	{ //RAW8 QVGA 16BPP * 3 phase + 16BPP * 3 AB
 		.width = 1536,
 		.height = 1920,
 		.pixel_rate = 488000000,
-		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
+	{ //RAW8 QVGA 16BPP * 3 phase
+		.width = 512,
+		.height = 1280,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
+	{ //RAW8 ADSD3030 * 3 phase
+		.width = 1024,
+		.height = 2080,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
-	{ //RAW16 QVGA 3 phase
+	{ //RAW8 ADSD3030 16BPP
 		.width = 512,
+		.height = 1040,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
+	{ //RAW8 ADSD3030 16BPP
+		.width = 1024,
+		.height = 1664,
+		.pixel_rate = 488000000,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
+		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
+	},
+	{ //RAW8 ADSD3030 16BPP
+		.width = 1024,
 		.height = 1280,
 		.pixel_rate = 488000000,
-		.code = MEDIA_BUS_FMT_SBGGR16_1X16,
+		.code = MEDIA_BUS_FMT_SBGGR8_1X8,
 		.link_freq_idx = 0 /* an index in link_freq_tbl[] */
 	},
+
 };
 
 static int adsd3500_regmap_write(void *context, const void *data, size_t count)
-- 
2.28.0

